### PR TITLE
zarajoy/c01-aws01

### PR DESCRIPTION
**Pull request by zarajoy**
# Description

Exercise: c01-aws01 

Student Name: zarajoy

> Exercises for c01-aws01

## Type of change

- [X] Exercise Submission (one exercise per PR)
  - [X] This PR is just for **one** exercise of a class
  - [X] PR name follows the pattern `<github-username>/<exercise-number>`
  - [X] Added exercise files inside folder `/classes/<class- number>/exercises/<exercise-number>/<github-username>/`
	- [ ] Review [exercise submission steps](./README.md#Exercises)
- [ ] Class content (instructors/admins)
- [ ] Documentation update
- [ ] Repo management

### CODE REVIEW

File: classes/01class/exercises/c01-aws01/zarajoy/c01-aws01.md [hunk 1]
```diff
@@ -0,0 +1,398 @@
...
+            "KeyName": "aws01kp",
+            "Tags": []
+        }
+    ]
+}
+z@bacon:~$ aws ec2 create-security-group --group-name my-sg --description "My security group" 
+{
+    "GroupId": "sg-0aa6a164d5faaf958"
+}
+z@bacon:~$ aws ec2 authorize-security-group-ingress --group-name my-sg --protocol tcp --port 22 --cidr 180.150.38.8/32
```
Comment (kikobr82) [inline classes/01class/exercises/c01-aws01/zarajoy/c01-aws01.md:21]: nice use of least privilege, giving only the access to your IP.. :)

File: classes/01class/exercises/c01-aws01/zarajoy/c01-aws01.md [hunk 2]
```diff
@@ -0,0 +1,398 @@
...
+- Create another EC2, also using the CLI instance in the same VPC but with private address only and connect to it.
+- Commands to create the second EC2 instance and any additional resource required:
+z@bacon:~$ aws ec2 create-key-pair --key-name paws01kp --query 'KeyMaterial' --output text > paws01kp.pem
+z@bacon:~$ sudo chmod 400 paws01kp.pem
+[sudo] password for z: 
+z@bacon:~$ aws ec2 create-security-group --group-name SecureGroup --description  "Secure group for DOA"
+{
+    "GroupId": "sg-09c8bcfe909954f1a"
+}
+z@bacon:~$ aws ec2 authorize-security-group-ingress --group-name SecureGroup --protocol tcp --port 22 --cidr 180.150.38.8/32
```
Comment (kikobr82) [inline classes/01class/exercises/c01-aws01/zarajoy/c01-aws01.md:213]: can you confirm that the security group for the instance without public IP only have this rule attached to it?

File: classes/01class/exercises/c01-aws01/zarajoy/c01-aws01.md [hunk 3]
```diff
@@ -0,0 +1,398 @@
...
+    ],
+    "OwnerId": "512742231244",
+    "ReservationId": "r-05d1863c7d002fa30"
+}
+- Commands to connect to the second EC2 instance:
+z@bacon:~$ eval 'ssh-agent'
+SSH_AUTH_SOCK=/tmp/ssh-22FZ7c7j6uMf/agent.203040; export SSH_AUTH_SOCK;
+SSH_AGENT_PID=203041; export SSH_AGENT_PID;
+echo Agent pid 203041;
+z@bacon:~$ aws ec2 describe-instances --query "Reservations[].Instances[].[InstanceId, PublicIpAddress]"
```
Comment (kikobr82) [inline classes/01class/exercises/c01-aws01/zarajoy/c01-aws01.md:362]: the second instance doesn't have a public IP address attached.

File: classes/01class/exercises/c01-aws01/zarajoy/c01-aws01.md (hunk 2 above)
Comment (zarajoy) [reply classes/01class/exercises/c01-aws01/zarajoy/c01-aws01.md:213]: z@bacon:~$ aws ec2 describe-instances --instance-ids i-08b574c6436ac7ad8
{
    "Reservations": [
        {
            "Groups": [],
            "Instances": [
                {
                    "AmiLaunchIndex": 0,
                    "ImageId": "ami-088ff0e3bde7b3fdf",
                    "InstanceId": "i-08b574c6436ac7ad8",
                    "InstanceType": "t2.micro",
                    "KeyName": "paws01kp",
                    "LaunchTime": "2020-06-16T06:43:58.000Z",
                    "Monitoring": {
                        "State": "disabled"
                    },
                    "Placement": {
                        "AvailabilityZone": "ap-southeast-2a",
                        "GroupName": "",
                        "Tenancy": "default"
                    },
                    "PrivateDnsName": "ip-172-31-45-4.ap-southeast-2.compute.internal",
                    "PrivateIpAddress": "172.31.45.4",
                    "ProductCodes": [],
                    "PublicDnsName": "",
                    "State": {
                        "Code": 16,
                        "Name": "running"
                    },
                    "StateTransitionReason": "",
                    "SubnetId": "subnet-48638000",
                    "VpcId": "vpc-ce545da9",
                    "Architecture": "x86_64",
                    "BlockDeviceMappings": [
                        {
                            "DeviceName": "/dev/xvda",
                            "Ebs": {
                                "AttachTime": "2020-06-16T06:43:59.000Z",
                                "DeleteOnTermination": true,
                                "Status": "attached",
                                "VolumeId": "vol-0338c6387c6e221c4"
                            }
                        }
                    ],
                    "ClientToken": "",
                    "EbsOptimized": false,
                    "EnaSupport": true,
                    "Hypervisor": "xen",
                    "NetworkInterfaces": [
                        {
                            "Attachment": {
                                "AttachTime": "2020-06-16T06:43:58.000Z",
                                "AttachmentId": "eni-attach-0f2eb3e32c4e8fde4",
                                "DeleteOnTermination": true,
                                "DeviceIndex": 0,
                                "Status": "attached"
                            },
                            "Description": "",
                            "Groups": [
                                {
                                    "GroupName": "SecureGroup",
                                    "GroupId": "sg-09c8bcfe909954f1a"
                                }
                            ],
                            "Ipv6Addresses": [],
                            "MacAddress": "06:87:b3:bd:c3:20",
                            "NetworkInterfaceId": "eni-01e2e4d50fd097c6b",
                            "OwnerId": "512742231244",
                            "PrivateDnsName": "ip-172-31-45-4.ap-southeast-2.compute.internal",
                            "PrivateIpAddress": "172.31.45.4",
                            "PrivateIpAddresses": [
                                {
                                    "Primary": true,
                                    "PrivateDnsName": "ip-172-31-45-4.ap-southeast-2.compute.internal",
                                    "PrivateIpAddress": "172.31.45.4"
                                }
                            ],
                            "SourceDestCheck": true,
                            "Status": "in-use",
                            "SubnetId": "subnet-48638000",
                            "VpcId": "vpc-ce545da9",
                            "InterfaceType": "interface"
                        }
                    ],
                    "RootDeviceName": "/dev/xvda",
                    "RootDeviceType": "ebs",
                    "SecurityGroups": [
                        {
                            "GroupName": "SecureGroup",
                            "GroupId": "sg-09c8bcfe909954f1a"
                        }
                    ],
                    "SourceDestCheck": true,
                    "VirtualizationType": "hvm",
                    "CpuOptions": {
                        "CoreCount": 1,
                        "ThreadsPerCore": 1
                    },
                    "CapacityReservationSpecification": {
                        "CapacityReservationPreference": "open"
                    },
                    "HibernationOptions": {
                        "Configured": false
                    },
                    "MetadataOptions": {
                        "State": "applied",
                        "HttpTokens": "optional",
                        "HttpPutResponseHopLimit": 1,
                        "HttpEndpoint": "enabled"
                    }
                }
            ],
            "OwnerId": "512742231244",
            "ReservationId": "r-05d1863c7d002fa30"
        }
    ]
}

Comment (zarajoy) [reply classes/01class/exercises/c01-aws01/zarajoy/c01-aws01.md:213]: cant work out a command to just show instanceID and groups associated :/

File: classes/01class/exercises/c01-aws01/zarajoy/c01-aws01.md (hunk 3 above)
Comment (zarajoy) [reply classes/01class/exercises/c01-aws01/zarajoy/c01-aws01.md:362]: correct. only the one from the first instance does.
    [
        "i-08b574c6436ac7ad8",
        null
    ],

**Comment by zarajoy** [comment]
Have renamed file to README.md, 
added ''' to break up code
shown secure group is only applied to 2nd instance.

File: classes/01class/exercises/c01-aws01/zarajoy/README.md [hunk 4]
```diff
@@ -0,0 +1,393 @@
...
+        {
+            "Description": "Secure group for DOA",
+            "GroupName": "SecureGroupv2",
+            "IpPermissions": [
+                {
+                    "FromPort": 22,
+                    "IpProtocol": "tcp",
+                    "IpRanges": [
+                        {
+                            "CidrIp": "172.31.36.157/32"
```
Comment (kikobr82) [inline classes/01class/exercises/c01-aws01/zarajoy/README.md:227]: Nice use of least privilege. For security groups, you can also configure the source as another Security group, so you could also configure this security group to allow only connects on port 22 from the Security group of the public instance and you would have a similar effect in this case.

**Pull request merged**